loki/transformations/transpile/tests/test_transpile.py

```diff
@@ -1249,6 +1251,77 @@ def test_transpile_multiconditional_range(tmp_path, frontend):
     with pytest.raises(NotImplementedError):
         f2c.apply(source=routine, path=tmp_path)
 
+
+@pytest.mark.parametrize('frontend', available_frontends())
+@pytest.mark.parametrize('dtype', ('integer', 'real',))
+@pytest.mark.parametrize('add_float', (False, True))
+def test_transpile_special_functions(tmp_path, builder, frontend, dtype, add_float):
+    """
+    A simple test to verify multiconditionals/select case statements.
+    """
+
+    fcode = f"""
+subroutine transpile_special_functions(in, out)
+  use iso_fortran_env, only: real64
+  implicit none
+  {dtype}{'(kind=real64)' if dtype == 'real' else ''}, intent(in) :: in
+  {dtype}{'(kind=real64)' if dtype == 'real' else ''}, intent(inout) :: out
+
+  if (mod(in{'+ 2._real64' if add_float else ''}, 2{'._real64' if dtype == 'real' else ''}{'+ 0._real64' if add_float else ''}) .eq. 0) then
+    out = 42{'._real64' if dtype == 'real' else ''}
+  else
+    out = 11{'._real64' if dtype == 'real' else ''}
+  endif
+end subroutine transpile_special_functions
```

Comment:
A suggestion for a slightly more readable code:

```suggestion
    if dtype == 'real':
        decl_type = f'{dtype}(kind=real64)'
        kind = f'._real64'
    else:
        decl_type = dtype
        kind = ''
     
    fcode = f"""
subroutine transpile_special_functions(in, out)
  use iso_fortran_env, only: real64
  implicit none
  {decl_type}, intent(in) :: in
  {decl_type}, intent(inout) :: out

  if (mod(in{'+ 2._real64' if add_float else ''}, 2{kind}{'+ 0._real64' if add_float else ''}) .eq. 0) then
    out = 42{kind}
  else
    out = 11{kind}
  endif
end subroutine transpile_special_functions
```